### PR TITLE
remove credits screen, restore original demo loop

### PR DESCRIPTION
**Pull request by rfomin**

### CODE REVIEW

File: src/d_main.c [hunk 1]
```diff
@@ -529,132 +511,107 @@ void D_AdvanceDemo (void)
...
+    else
     {
-      demosequence = 0;
-      name = demostates[demosequence][gamemode].name;
+        demosequence = (demosequence + 1) % 6;
     }
-  }
 
-  demostates[demosequence][gamemode].func(name);
+    switch (demosequence)
```
Comment (fabiangreffrath) [inline src/d_main.c:545]: Is this from Chocolate Doom?

Comment (rfomin) [reply src/d_main.c:545]: Yes.

**Comment by liPillON** [comment]
speaking of demoloops... 
what's your opinion on introducing a user setting to prevent them from playing?
I was thinking about submitting a feature request..

**Comment by MrAlaux** [comment]
> speaking of demoloops...
> what's your opinion on introducing a user setting to prevent them from playing?

Had I told you that you can use the Pause key in the menu to stop them? Not to be insistent, I'm just genuinely unsure if I told you in particular.

**Comment by rfomin** [comment]
> what's your opinion on introducing a user setting to prevent them from playing?

I think we can add a config-only option.

**Comment by fabiangreffrath** [comment]
You can create four empty files called `demo1.lmp` to `demo4.lmp` and place them in your `autoload/all-all` directory.

**Comment by liPillON** [comment]
yeah I know about that trick (it works only for iwads though) 

I came up with this request just after trying the updated aaliens pwad, which has on top of the qol changes has been also converted to mbf21 - so the embedded demos probably desync

**Comment by fabiangreffrath** [comment]
> has been also converted to mbf21 - so the embedded demos probably desync

Why should the demos desync? We support all relevant complevels up to and including mbf21.

**Comment by rfomin** [comment]
> I came up with this request just after trying the updated aaliens pwad, which has on top of the qol changes has been also converted to mbf21

Interesting, is this update is public?

> Why should the demos desync? We support all relevant complevels up to and including mbf21.

The internal demos in aaliens.wad are comlevel boom, they probably desync if there are any changes in levels/dehacked.

**Comment by rfomin** [comment]
Found it in Skillsaw profile: https://www.doomworld.com/profile/8877-skillsaw/ Beautiful!

**Review by fabiangreffrath — APPROVED** [review]
Okay, I buy that going back to the Vanilla demo sequence is a net win. And that sequence does not have a credits screen. So, with a tear in my eye, I approve the removal of the credits screen.

**Comment by rfomin** [comment]
> So, with a tear in my eye, I approve the removal of the credits screen.

Thank you!

**Pull request merged**